content/en/docs/components/spark-operator/user-guide/writing-sparkapplication.md

```diff
@@ -0,0 +1,647 @@
+
+## Using Tolerations
+
+A `SparkApplication` can specify an `Tolerations` for the driver or executor pod, using the optional field `.spec.driver.tolerations` or `.spec.executor.tolerations`. Below is an example:
+
+```yaml
+spec:
+  driver:
+    tolerations:
+    - key: Key
+      operator: Exists
+      effect: NoSchedule
+
+  executor:
+    tolerations:
+    - key: Key
+      operator: Equal
+      value: Value
+      effect: NoSchedule
+```
+
+Note that the mutating admission webhook is needed to use this feature. Please refer to the
+[Getting Started](../../getting-started) on how to enable the mutating admission webhook.
+
+## Using Security Context
+
+A `SparkApplication` can specify a `SecurityContext` for the driver or executor containers, using the optional field `.spec.driver.securityContext` or `.spec.executor.securityContext`.
+`SparkApplication` can also specify a `PodSecurityContext` for the driver or executor pod, using the optional field `.spec.driver.podSecurityContext` or `.spec.executor.podSecurityContext`. Below is an example:
+
+```yaml
+spec:
+  driver:
+    podSecurityContext:
+      runAsUser: 1000
+    securityContext:
+      allowPrivilegeEscalation: false
+      runAsUser: 2000
+  executor:
+    podSecurityContext:
+      runAsUser: 1000
+    securityContext:
+      allowPrivilegeEscalation: false
+      runAsUser: 2000
+```
+
+Note that the mutating admission webhook is needed to use this feature. Please refer to the
+[Getting Started](../../getting-started) on how to enable the mutating admission webhook.
+
+## Using Sidecar Containers
+
+A `SparkApplication` can specify one or more optional sidecar containers for the driver or executor pod, using the optional field `.spec.driver.sidecars` or `.spec.executor.sidecars`. The specification of each sidecar container follows the [Container](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#container-v1-core) API definition. Below is an example:
+
+```yaml
+spec:
+  driver:
+    sidecars:
+    - name: "sidecar1"
+      image: "sidecar1:latest"
+      ...
+  executor:
+    sidecars:
+    - name: "sidecar1"
+      image: "sidecar1:latest"
+      ...
+```
+
+Note that the mutating admission webhook is needed to use this feature. Please refer to the
+[Getting Started](../../getting-started) on how to enable the mutating admission webhook.
+
+## Using Init-Containers
+
+A `SparkApplication` can optionally specify one or more [init-containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) for the driver or executor pod, using the optional field `.spec.driver.initContainers` or `.spec.executor.initContainers`, respectively. The specification of each init-container follows the [Container](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#container-v1-core) API definition. Below is an example:
```

Comment:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#container-v1-core is broken.